cmd/root.go

```diff
@@ -439,13 +434,25 @@ func watchFile(ctx context.Context, toolsFileName string, s *server.Server) {
 
 	defer w.Close()
 
-	err = w.Add(filepath.Dir(toolsFileName))
-	if err != nil {
-		logger.WarnContext(ctx, "error adding the tools file to watcher %s", err)
+	watchingFolder := false
+	var folderToWatch string
+
+	// if watchedFiles is empty, indicates that user passed entire folder instead
+	if len(watchedFiles) == 0 {
+		watchingFolder = true
+		for k := range watchDirs {
+			folderToWatch = k
+		}
 	}
 
-	cleanedFilename := filepath.Clean(toolsFileName)
-	logger.DebugContext(ctx, fmt.Sprintf("Now watching tools file %s", cleanedFilename))
+	for dir := range watchDirs {
+		err := w.Add(dir)
+		if err != nil {
+			logger.WarnContext(ctx, fmt.Sprintf("Error adding path %s to watcher: %s", dir, err))
+		} else {
+			logger.DebugContext(ctx, fmt.Sprintf("Added directory %s to watcher.", dir))
+		}
```

Comment:
nit: happy path to the left
```suggestion
		if err != nil {
			logger.WarnContext(ctx, fmt.Sprintf("Error adding path %s to watcher: %s", dir, err))
		}
		logger.DebugContext(ctx, fmt.Sprintf("Added directory %s to watcher.", dir))

```